cmd/infra/powervs/destroy.go

```diff
@@ -371,8 +376,64 @@ func deleteSecrets(name, namespace, cloudInstanceID string, accountID string, re
 	return nil
 }
 
+// destroyPowerVsDhcpServer destroying powervs dhcp server
+func destroyPowerVsDhcpServer(ctx context.Context, logger logr.Logger, cloudInstanceID string, session *ibmpisession.IBMPISession) error {
+	client := instance.NewIBMPIDhcpClient(ctx, session, cloudInstanceID)
+
+	dhcpServers, err := client.GetAll()
+	if err != nil {
+		return err
+	}
+
+	if len(dhcpServers) == 0 {
+		logger.Info("No DHCP servers available to delete in PowerVS")
+		return nil
+	}
+
+	dhcpID := *dhcpServers[0].ID
+	logger.Info("Deleting DHCP server", "id", dhcpID)
+	err = client.Delete(dhcpID)
+	if err != nil {
+		return err
+	}
+
+	instanceClient := instance.NewIBMPIInstanceClient(ctx, session, cloudInstanceID)
+
+	// TO-DO: need to replace the logic of waiting for dhcp service deletion by using jobReference.
+	// jobReference is not yet added in SDK
+	f := func() (bool, error) {
+		dhcpInstance, err := instanceClient.Get(dhcpID)
+		if err != nil {
+			if err = isNotRetryableError(err, timeoutErrorKeywords); err == nil {
+				return false, nil
+			}
+			errMsg := err.Error()
+			// when instance becomes does not exist, infra destroy can proceed
+			// when there is flaky, internal server error in pvm-instance get call, retry destroy
+			if strings.Contains(errMsg, "pvm-instance does not exist") || strings.Contains(errMsg, "unable to get the server volume attachment list: Internal Server Error") {
```

Comment:
```suggestion
			if strings.Contains(errMsg, "pvm-instance does not exist") || strings.Contains(errMsg, "unable to get the server volume attachment list") {
```

Reply:
You are returning `true, nil`, how can it retry when it gets this flaky error?

Reply:
Rectified it, thanks!